activesupport/test/core_ext/enumerable_test.rb

```diff
@@ -241,6 +241,24 @@ def test_pluck
       ExpandedPayment.new(10, 50)
     ])
     assert_equal [[5, 99], [15, 0], [10, 50]], payments.pluck(:dollars, :cents)
+
+    assert_equal [], [].pluck(:price)
+    assert_equal [], [].pluck(:dollars, :cents)
+  end
+
+  def test_pick
+    payments = GenericEnumerable.new([ Payment.new(5), Payment.new(15), Payment.new(10) ])
+    assert_equal 5, payments.pick(:price)
+
+    payments = GenericEnumerable.new([
+      ExpandedPayment.new(5, 99),
+      ExpandedPayment.new(15, 0),
+      ExpandedPayment.new(10, 50)
+    ])
+    assert_equal [5, 99], payments.pick(:dollars, :cents)
+
+    assert_nil [].pick(:price)
+    assert_equal [nil, nil], [].pick(:dollars, :cents)
```

Comment:
Should we normalize to return an empty array like pluck does? Could potentially tap compact! at the end of the map.

Reply:
I don't think we can use compact, since we need to keep any nil values that were picked when the enumerable isn't empty. We could add a special case for when it _is_ empty, though.

Using the rubric that `pick(...)` should be equivalent to `pluck(...).first`, picking an empty enumerable seems like it should return nil:

```ruby
irb(main):001:0> [].pluck(:dollar, :cents).first
=> nil
```

This is how `Relation#pick` already behaves:

https://github.com/rails/rails/blob/f265e0ddb1139a91635b7905aae1be76b22c6db1/activerecord/test/cases/calculations_test.rb#L917

I have a slight preference for aligning with that existing behaviour, although in practice I think the result of calling `pick` with multiple arguments is usually destructured, so it doesn't matter too much which behaviour we ~pick~ choose:

```ruby
irb(main):001:0> a, b = nil; [a, b]
=> [nil, nil]
irb(main):002:0> a, b = []; [a, b]
=> [nil, nil]
irb(main):003:0> a, b = [nil, nil]; [a, b]
=> [nil, nil]
```

Reply:
My objection was more to seeing `[]` for the `pluck` return versus `[ nil, nil ]` for `pick`, size would be different or similar.

> I have a slight preference for aligning with that existing behaviour,

Same, but the nil is the default return for a method call, so it would do that regardless. I like the `none?` fix.